### PR TITLE
fix: facing multiple issue with multiple stripe account feature #4781

### LINKED ISSUE #4781: fix: facing multiple issues with 2.7.0 multiple stripe account feature
# Bug Report

## User Story

As an admin, I want multiple stripe account feature to work fine so that setup it easily and accept the donation.

## Details

I am facing multiple issues:
- [x] Update and Edit account name for the manual key does not work fine.
image

- [x] I get the option to add a manual stripe key even if this feature disable.
image

- [x] Do not show an alert when updating the manual stripe key.
image

- [x]  Do not show an alert when disconnecting the manual stripeaccount.
image

- [x] Always keep the first manual key account as default when adding only manual keys
image

- [x] Styling issue.
image

- [x] Can you confirm that name will be attached to a manual key account because I can not see names.
image
image

- [x] Show globally selected default account as default account in per-form settings.
image
image

- [x] Store Stripe account information like name in donation data for future reference: https://github.com/impress-org/givewp/pull/4662#issuecomment-623706801

## Expected Behavior

Multiple Stripe account feature must be bug-free :D

## Bug Type

These bugs only exist in the upcoming Give 2.7.0.

## Acceptance Criteria

- [x] The Issues mentioned in `Details` section have been fixed and functionality is working as expected.

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR resolves #4781 

## What to test
I have tested the changes as per the acceptance criteria on the issue description.

## Screenshots:
image

image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@mehul0810 I am getting this issue when adding `Manual` type stripe account
```
[03-Jun-2020 13:47:20 UTC] PHP Notice:  Undefined index: account_name in /Users/ravinderkumar/Local-Sites/give/app/public/wp-content/plugins/give-stripe/includes/admin/give-stripe-settings.php on line 417
[03-Jun-2020 13:47:20 UTC] PHP Notice:  Undefined index: account_email in /Users/ravinderkumar/Local-Sites/give/app/public/wp-content/plugins/give-stripe/includes/admin/give-stripe-settings.php on line 420
[03-Jun-2020 13:47:20 UTC] PHP Notice:  Undefined index: give_stripe_user_id in /Users/ravinderkumar/Local-Sites/give/app/public/wp-content/plugins/give-stripe/includes/admin/give-stripe-settings.php on line 431
[03-Jun-2020 13:47:20 UTC] PHP  16. do_action() /Users/ravinderkumar/Local-Sites/give/app/public/wp-content/plugins/give/includes/admin/class-admin-settings.php:991
```

**Comment by mehul0810** [comment]
@ravinderk The requested minor changes are fixed. Please review

**Comment by ravinderk** [comment]
@mehul0810 please do testing carefully fully:

- [ ] I am not able to see `Update` and `Cancel` link when clicking on the `Edit` link for manual key stripe account if it set to the `default` account.

image

File: assets/src/js/admin/stripe-admin.js [hunk 1]
```diff
@@ -352,17 +372,20 @@ window.addEventListener( 'DOMContentLoaded', function() {
...
 						modalWrapper: 'give-modal--warning',
 					},
 					modalContent: {
 						title: Give.fn.getGlobalVar( 'disconnect_stripe_title' ),
-						desc: e.target.getAttribute( 'data-disconnect-message' ),
+						desc: currentElement.getAttribute( 'data-disconnect-message' ),
 					},
 					successConfirm: () => {
-						window.location.href = e.target.getAttribute( 'href' );
+						console.log( `${ currentElement.getAttribute( 'href' ) }&account=${ currentElement.getAttribute( 'data-account' ) }` );
```
Comment (ravinderk) [inline assets/src/js/admin/stripe-admin.js:387]: @mehul0810 did you left this code intentionally?

**Comment by ravinderk** [comment]
@mehul0810  I find the following issues when testing this pr:

- [ ] I am getting PHP notice when saving stripe account (make sure you at least have on manual type stripe account).

```
[03-Jun-2020 17:03:36 UTC] PHP Notice:  Undefined index: give_stripe_user_id in /Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/class-wp-list-util.php on line 155
[03-Jun-2020 17:03:36 UTC] PHP Stack trace:
[03-Jun-2020 17:03:36 UTC] PHP   1. {main}() /Users/ravinderkumar/Local-Sites/give/app/public/wp-admin/edit.php:0
[03-Jun-2020 17:03:36 UTC] PHP   2. require_once() /Users/ravinderkumar/Local-Sites/give/app/public/wp-admin/edit.php:10
[03-Jun-2020 17:03:36 UTC] PHP   3. do_action() /Users/ravinderkumar/Local-Sites/give/app/public/wp-admin/admin.php:170
[03-Jun-2020 17:03:36 UTC] PHP   4. WP_Hook->do_action() /Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/plugin.php:478
[03-Jun-2020 17:03:36 UTC] PHP   5. WP_Hook->apply_filters() /Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/class-wp-hook.php:312
[03-Jun-2020 17:03:36 UTC] PHP   6. call_user_func_array:{/Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/class-wp-hook.php:288}() /Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/class-wp-hook.php:288
[03-Jun-2020 17:03:36 UTC] PHP   7. give_stripe_connect_save_options() /Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/class-wp-hook.php:288
[03-Jun-2020 17:03:36 UTC] PHP   8. wp_list_pluck() /Users/ravinderkumar/Local-Sites/give/app/public/wp-content/plugins/give/includes/gateways/stripe/includes/admin/admin-actions.php:60
[03-Jun-2020 17:03:36 UTC] PHP   9. WP_List_Util->pluck() /Users/ravinderkumar/Local-Sites/give/app/public/wp-includes/functions.php:4470
```

As I suggested you keep data modal same for each account but it will keep time taking reacting at this time but inconsistency in the data model of the cause of this:

image

```php
<?php
	// If the same Stripe account is already connected, then don't save it.
	if ( in_array( $stripe_account_id, wp_list_pluck( $stripe_accounts, 'give_stripe_user_id' ), true ) ) {
		return;
	}
```
As you can see above code, you are trying to get same key from each array but all stripe account may not have this array key.

**Comment by mehul0810** [comment]
@ravinderk I have updated code and reviewed both the PRs https://github.com/impress-org/givewp/pull/4785 and https://github.com/impress-org/give-stripe/pull/451

Also, all the above mentioned points and the points related to the legacy code has been removed.

You can take a look on the short video: https://www.loom.com/share/5bd0e5047ef444319ce7ea39daf45c61

Let me know if you have any questions while testing.

Thanks!

**Pull request merged**